### ISSUE TITLE
Creating placeholder with `np.uint32` dtype fails

### ISSUE THREAD

**Issue by SpikingNeurons** [tensorflow/tensorflow#16407]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
yes custom snippet below
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 7
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.4.0
- **Python version**: 
3.6
- **Bazel version (if compiling from source)**:
not applicable
- **GCC/Compiler version (if compiling from source)**:
not applicable
- **CUDA/cuDNN version**:
CUDA 8.5
- **GPU model and memory**:
Titan X
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
np.uint32 dtype is not supported while creating placeholder

### Source code / logs
```python
import tensorflow as tf
import numpy as np
print(tf.placeholder(np.int32, [None], 'ph1'))
print(tf.placeholder(np.uint32, [None], 'ph2'))
```
Line 3 works, line 4 fails.

Console Output:
```txt
Tensor("ph1:0", shape=(?,), dtype=int32)
Traceback (most recent call last):
  File "...\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\eager\execute.py", line 126, in make_type
    v = dtypes.as_dtype(v).base_dtype
  File "...\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\framework\dtypes.py", line 595, in as_dtype
    "Cannot convert value %r to a TensorFlow DType." % type_value)
TypeError: Cannot convert value <class 'numpy.uint32'> to a TensorFlow DType.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:/scaffold-ext/scaffold_ext/analysis/ann/bug.py", line 4, in <module>
    print(tf.placeholder(np.uint32, [None], 'ph2'))
  File "...\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 1599, in placeholder
    return gen_array_ops._placeholder(dtype=dtype, shape=shape, name=name)
  File "...\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 3083, in _placeholder
    dtype = _execute.make_type(dtype, "dtype")
  File "...\AppData\Local\Continuum\anaconda3\lib\site-packages\tensorflow\python\eager\execute.py", line 129, in make_type
    (arg_name, repr(v)))
TypeError: Expected DataType for argument 'dtype' not <class 'numpy.uint32'>.
```

**Comment by quaeler** [comment]
This has been addressed since the 1.4.0 release:
https://github.com/tensorflow/tensorflow/commit/3bafe0a86f67dd54197c6d60bdb5053f510de7d8#diff-85e323da48079f7831b7e9f6ec24cda2R501

Please give 1.5.0 a shot.

**Comment by SpikingNeurons** [comment]
Okay, I was thinking so.
